### PR TITLE
Hotfix/non linear q

### PR DESCRIPTION
**Pull request by jduerholt**
## Motivation

When implementing the possibility to use `q>1` for nonlinear constraints in PR https://github.com/pytorch/botorch/pull/1793, I forgot to update the `batch_limit` check when commenting the check back in. Thus, the feature is still not usable. Here, it is fixed and it is actually tested that it is working with `q>1`.

Before, only the lower level methods were tested with `q>1`, but not `optimize_acqf`. This demonstrates again, that one cannot write to less tests :D

cc: @Balandat

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Unit tests.

### CODE REVIEW

File: botorch/optim/optimize.py
Comment (Balandat) [inline botorch/optim/optimize.py:1042]: 🙃

File: test/optim/test_optimize.py
Comment (Balandat) [inline test/optim/test_optimize.py:913]: nit
```suggestion
                        torch.tensor([[1.0, 1.0, 2.0]], **tkwargs),
```

File: test/optim/test_optimize.py
Comment (Balandat) [inline test/optim/test_optimize.py:912]:
```suggestion
            for candidate in candidates:
                self.assertTrue(
                    torch.allclose(
                        torch.sort(candidate).values,
```

**Review by Balandat — APPROVED** [review]
Thanks for the fix! A couple of µ-nits, o/w this lgtm.

**Comment by jduerholt** [comment]
@Balandat: thanks for the quick review! I just updated according to your suggestions. 

Btw: are you changing the contributors repository permissions? Since today, when creating a PR or updating it, the test pipelines are not starting anymore automatically but are waiting for a maintainer approval ...

**Comment by Balandat** [comment]
> Btw: are you changing the contributors repository permissions? Since today, when creating a PR or updating it, the test pipelines are not starting anymore automatically but are waiting for a maintainer approval ...

Yeah I noticed that as well. Not sure what's going on, I don't know of any changes that we made on our end. The default is that only first-time contributors need approval, once they've merged code they should be good. Maybe for some reason the permissions got reset? Let's see if this still is an issue for the next contribution (once this one gets merged).

**Comment by Balandat** [comment]
> Yeah I noticed that as well. Not sure what's going on, I don't know of any changes that we made on our end. The default is that only first-time contributors need approval, once they've merged code they should be good. Maybe for some reason the permissions got reset? Let's see if this still is an issue for the next contribution (once this one gets merged).

Turns out that was intended by the pytorch org owners in order to tighten up security on the repos - this is intended to be temporary though and should go back to the previous behavior.

**Pull request closed without merging**
